corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/mascReader/MascXCESReader.java

```diff
@@ -0,0 +1,298 @@
+                        int endToken = (int)coveredTokenNodes.item(coveredTokenNodes.getLength() - 1)
+                                .getUserData(TOKEN_IDENTIFIER_KEY) + 1;
+                        spanLabels.get(processor.getViewName()).add(new Pair<>(new IntPair(beginToken, endToken), label));
+                    }
+                }
+            }
+        }
+
+        TextAnnotation ta = BasicTextAnnotationBuilder.createTextAnnotationFromTokens(
+                corpusName, textId, Collections.singletonList(tokens.toArray(new String[0])));
+
+        for (TokenLabelProcessor processor : TOKEN_LABEL_PROCESSORS) {
+            createTokenLabelView(tokenLabels.get(processor.getViewName()).stream(), ta, processor.getViewName());
+        }
+
+        for (SpanLabelProcessor processor : SPAN_LABEL_PROCESSORS) {
+            createSpanLabelView(spanLabels.get(processor.getViewName()).stream(), ta, processor.getViewName(), true);  // span labels in MASC dataset might overlap
+        }
+
+        return ta;
+    }
+
+    /**
+     * Helper for create a processor function that take a particular attribute of the element as the label
+     */
+    private static Function<Element, String> simpleAttrProcessor(String attr) {
+        return elem -> Optional
+                .ofNullable(elem.getAttributes().getNamedItem(attr))  // the label is the `attr` attribute of the element
+                .map(Node::getNodeValue)
+                .orElse(null);  // or don't create the label if the attribute doesn't exist
+    }
+
+    /**
+     * Helper for create a TokenLabelView from a stream of token labels
+     */
+    private static int createTokenLabelView(
+            Stream<Pair<Integer, String>> tokenLabels,
+            TextAnnotation ta,
+            String viewName) {
+        TokenLabelView view = new TokenLabelView(viewName, "GoldStandard", ta, 1.0);
+        tokenLabels.forEach(label -> view.addTokenLabel(label.getFirst(), label.getSecond(), 1.0));
+        ta.addView(viewName, view);
+        return view.count();
+    }
+
+    /**
+     * Helper for create a SpanLabelView from a stream of span labels
+     */
+    private static int createSpanLabelView(
+            Stream<Pair<IntPair, String>> spans,
+            TextAnnotation ta,
+            String viewName,
+            boolean allowOverlapping) {
+        SpanLabelView view = new SpanLabelView(viewName, "GoldStandard", ta, 1.0, allowOverlapping);
+        spans.forEach(span -> view.addSpanLabel(
+                span.getFirst().getFirst(), span.getFirst().getSecond(), span.getSecond(), 1.0));
+        ta.addView(viewName, view);
+        return view.count();
+    }
+
+    @Override
+    protected void initializeReader() {}
+
+    @Override
+    public boolean hasNext() {
+        return textAnnotations.size() > currentAnnotationId;
+    }
+
+    @Override
+    public TextAnnotation next() {
+        return textAnnotations.get(currentAnnotationId++);
+    }
+
+    @Override
```

Comment:
if you catch any exceptions that indicate failure to process either a file or a portion of a file's content, please add the list of files and message text indicating that these are failures to the report you generate.